java/src/test/java/com/mlt/TestUtils.java

```diff
@@ -80,24 +85,49 @@ public static void compareTilesSequential(MapLibreTile mlTile, MapboxVectorTile
       for (com.mlt.data.Feature mvtFeature : mvtFeatures) {
         var mltFeature =
             mltFeatures.stream().filter(f -> f.id() == mvtFeature.id()).findFirst().get();
+
         assertEquals(mvtFeature.id(), mltFeature.id());
 
         var mltGeometry = mltFeature.geometry();
         var mvtGeometry = mvtFeature.geometry();
-        assertEquals(mvtGeometry, mltGeometry);
+        if (!mvtGeometry.toString().equals(mltGeometry.toString())) {
```

Comment:
Do you get any additonal error with this change?
The Geometry classes used for the equlity tests are from the JTS Topology Suite library, which does proper geometry equality comparison based on the specific geometry types and  is battle tested :) Since we are provding no tolerance values for the current equality comparision we are also not safe for floating point rounding errors like in the String variant but we could introduce a value for this to be more robust.
By using `toString` JTS converts it to WKT. 
What are your thoughts on this?

Reply:
Yes, I also tried with `if (!mvtGeometry.equals(mltGeometry)) {` and I get 7 exceptions like:

```
MltDecoderTest > Decode OMT Tiles > com.mlt.decoder.MltDecoderTest.decodeOMTTiles(String)[13] FAILED
    org.locationtech.jts.geom.TopologyException: side location conflict [ (2639.311475409836, 3217.967213114754, NaN) ]
        at app//org.locationtech.jts.geomgraph.EdgeEndStar.propagateSideLabels(EdgeEndStar.java:289)
        at app//org.locationtech.jts.geomgraph.EdgeEndStar.computeLabelling(EdgeEndStar.java:125)
        at app//org.locationtech.jts.operation.relate.RelateComputer.labelNodeEdges(RelateComputer.java:325)
        at app//org.locationtech.jts.operation.relate.RelateComputer.computeIM(RelateComputer.java:125)
        at app//org.locationtech.jts.operation.relate.RelateOp.getIntersectionMatrix(RelateOp.java:109)
        at app//org.locationtech.jts.operation.relate.RelateOp.relate(RelateOp.java:54)
        at app//org.locationtech.jts.geom.Geometry.relate(Geometry.java:1035)
        at app//org.locationtech.jts.geom.Geometry.equalsTopo(Geometry.java:1089)
        at app//org.locationtech.jts.geom.Geometry.equals(Geometry.java:1057)
        at app//com.mlt.TestUtils.compareTilesSequential(TestUtils.java:93)
        at app//com.mlt.decoder.MltDecoderTest.testTile(MltDecoderTest.java:158)
        at app//com.mlt.decoder.MltDecoderTest.decodeOMTTiles(MltDecoderTest.java:122)
```

Do you also see that locally?

Try changing from:

```
assertEquals(mvtGeometry, mltGeometry);
```

To:

```
import static org.junit.jupiter.api.Assertions.assertTrue;
assertTrue(mvtGeometry.equals(mltGeometry), "MVTGeom != MLTGeom");
```

Reply:
For now in this PR I've reverted this custom handling: `[cb75bdc](https://github.com/maplibre/maplibre-tile-spec/pull/168/commits/cb75bdc075ceec0d37dcfdf227177c657d41c42e)`

Let's discuss larger issue at https://github.com/maplibre/maplibre-tile-spec/issues/178